File: Frontend/src/app/model/comment.ts
import {User} from './user';
import {Ad} from './ad';
import {Car} from './car';

export class Comment {
  id: number;
  commenter: User;
  comment: string;
  approved: boolean;
  carRating: number;
  ad: Ad;
  cars: Array<Car> = new Array<Car>();

/*
  constructor(id?: number, commenter?: User, comment?: string, approved?: boolean, carRating?: number, ad?: Ad) {
    this.id = id;
    this.commenter = commenter;
    this.comment = comment;
    this.approved = approved;
    this.carRating = carRating;
    this.ad = ad;
  }*/


  constructor() {
  }
}

File: services/rentingService/src/main/java/service/rentingService/model/Comment.java
package service.rentingService.model;

import javax.persistence.*;

@Entity
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "commenter_id", nullable = false)
    private User commenter;

    @Column
    private String comment;

    @Column
    private boolean approved = false;

    @Column
    private int carRating;

    @ManyToOne
    @JoinColumn(name = "ad_id")
    private Ad ad;

    public Comment() {
    }

    public Comment(User commenter, String comment, boolean approved, int carRating, Ad ad) {
        this.commenter = commenter;
        this.comment = comment;
        this.approved = approved;
        this.carRating = carRating;
        this.ad = ad;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public User getCommenter() {
        return commenter;
    }

    public void setCommenter(User commenter) {
        this.commenter = commenter;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public boolean isApproved() {
        return approved;
    }

    public void setApproved(boolean approved) {
        this.approved = approved;
    }

    

    public Ad getAd() {
        return ad;
    }

    public void setAd(Ad ad) {
        this.ad = ad;
    }

    public int getCarRating() {
        return carRating;
    }

    public void setCarRating(int carRating) {
        this.carRating = carRating;
    }
}
